en/hardware/porting_guide_config.md

```diff
@@ -0,0 +1,46 @@
+# PX4 Board Configuration (Kconfig)
+
+The PX4 Autopilot firmware can be configured at build time to adapt it for specialized applications (fixedwing, multicopter, rover or more), to enable new and experimental features (such as RTPS or UAVCANv1) or to save flash & RAM usage by disabling some drivers and subsystems.
+This configuration is handled through *Kconfig*, which is the same [configuration system used by NuttX](../hardware/porting_guide_nuttx.md#nuttx-menuconfig-setup).
+
+The configuration options (often referred as "symbols" by the *kconfig* language) are defined in `Kconfig` files under the **/src** directory. 
+
+## PX4 Kconfig Symbol Naming Convention
+
+By convention, symbols for modules/drivers are named based on the module folder path.
+For example, the symbol for the ADC driver at `src/drivers/adc/board_adc` must be named `DRIVERS_ADC_BOARD_ADC`.
+
+Symbols for driver/module specific options should omit the path, use the name of the module, and depend on the symbol.
```

Comment:
What do you mean by "depend on the symbol"? I THINK from looking at the example that you mean that UAVCAN_V1_GNSS_PUBLISHER is a symbol for an option used in DRIVERS_UAVCAN_V1 - so `UAVCAN_V1` is the module name.

I find this a little confusing because while I get the above, and I can see that UAVCAN_V1_GNSS_PUBLISHER is conditional on `DRIVERS_UAVCAN_V1` due to the `if` statement, I don't understand what the example I am looking at belongs to - though presumably it is pat of the config file for uavcan v1

Reply:
The `UAVCAN_V1_GNSS_PUBLISHER` key is an example of an option for the `UAVCAN_V1` application. Thus you've to guard these options behind the `DRIVERS_UAVCAN_V1` key otherwise you end up with lots of options which aren't applicable if the application itself isn't enabled (thus the dependency).

I do agree it might be a bit confusing right now, would it help to remove the wording "depend" and replace it by conditional?

Reply:
I think it might help. But what I'd do is spell it out just as you have for me-

So something like (feel free to adjust if I was not correct):

----

Symbols for driver/module specific options should omit the path, use the name of the module, and are conditionally defined based on the module symbol. For example, below the UAVCAN_V1 module (in driver/uavcan_v1/?) with module symbol `DRIVERS_UAVCAN_V1` defines the option `UAVCAN_V1_GNSS_PUBLISHER` (but only if DRIVERS_UAVCAN_V1 exists).